### ISSUE TITLE
React - id value not present when linking through field on a relationship using JDL -> Entity{fk(field)}

### ISSUE THREAD

**Issue by delkant** [jhipster/generator-jhipster#7986]
##### **Overview of the issue**

No id present when updating an entity if the relationship was defined to be linked through field different from default(the ID)
i.e.(using name):
`relationship OneToMany {
  Author{book(name)} to Book{writer(name)}
}`
##### **Motivation for or Use Case**
Updating an existing value on DB.

Server will throw and exception since the id of the relationship(which exists) is not present. Even if you are changing other fields(leaving the relationship as it is)
##### **Reproduce the error**
Update a value on an existing record;

TransientPropertyValueException: object references an unsaved transient instance - save the transient instance before flushing

##### **Related issues**
Maybe: https://github.com/jhipster/generator-jhipster/issues/7967#issuecomment-405064024

`<AvInput id="blog-user" type="hidden" name="user.id" value={this.state.userId} />
`
5.1.0 did not generated above code. I added the code in 5.1.0 and works perfectly.

##### **JHipster Version(s)**
5.1.0

##### **JHipster configuration**

```
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.test",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.1.0",
    "applicationType": "gateway",
    "baseName": "PTasks",
    "packageName": "com.test",
    "packageFolder": "com/test",
    "serverPort": "9980",
    "authenticationType": "uaa",
    "uaaBaseName": "uaa",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "gradle",
    "enableSwaggerCodegen": false,
    "clientFramework": "react",
    "useSass": false,
    "clientPackageManager": "yarn",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "es"
    ]
  }
}
```
##### **Browsers and Operating System**

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by deepu105** [comment]
@agaspardcilia can you check this

**Comment by agaspardcilia** [comment]
It's supposed to be fixed with #7981. I'll make sure it's working properly tho.

**Comment by deepu105** [comment]
@agaspardcilia can you add fix #no for this as well on the ticket
